internal/controller/persistentvolumeclaim_controller.go

```diff
@@ -66,16 +58,81 @@ func (r *PersistentVolumeClaimReconciler) Reconcile(ctx context.Context, req ctr
 		}, nil
 	}
 
-	// Skip if the PVC is not deleted or PVC does not have TopoLVM's finalizer.
-	if pvc.DeletionTimestamp == nil {
+	// Skip if PVC does not have TopoLVM's finalizer.
+	if !controllerutil.ContainsFinalizer(pvc, topolvm.PVCFinalizer) {
 		return ctrl.Result{}, nil
 	}
 
-	if !controllerutil.ContainsFinalizer(pvc, topolvm.PVCFinalizer) {
-		return ctrl.Result{}, nil
+	if pvc.DeletionTimestamp == nil {
+		err := r.notifyKubeletToResizeFS(ctx, log, pvc)
+		return ctrl.Result{}, err
+	}
+
+	result, err := r.deletePodsUsingDeletingPVC(ctx, log, pvc)
+	if result.Requeue || err != nil {
+		return result, err
+	}
+
+	controllerutil.RemoveFinalizer(pvc, topolvm.PVCFinalizer)
+
+	if err := r.client.Update(ctx, pvc); err != nil {
+		log.Error(err, "failed to remove finalizer", "name", pvc.Name)
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *PersistentVolumeClaimReconciler) notifyKubeletToResizeFS(ctx context.Context, log logr.Logger, pvc *corev1.PersistentVolumeClaim) error {
```

Comment:
```suggestion
func (r *PersistentVolumeClaimReconciler) notifyKubeletToResizeFS(ctx context.Context, pvc *corev1.PersistentVolumeClaim) error {
logger := crlog.FromContext(ctx)
```

Reply:
I wrote it as you suggested, but I felt it was redundant.
Could you tell me why you prefer it?

Reply:
I think its best practice nowadays to only pass context and retrieve logger from context to have less parameter bloat, really just a nit that I picked up over the years. Feel free to change back if you feel strongly about it

Reply:
Ok. I adapt it to the style.